src/app/courses/courses-page/courses-page.component.ts

```diff
@@ -14,21 +25,37 @@ import { BreadCrumb } from '../breadcrumbs/breadcrumbs.component';
 export class CoursesPageComponent implements OnInit {
 
   textToSearch = '';
-  filterPipe = new FilterPipe();
+  public currentPageIndex = 1;
+  public totalCount = 0;
   public courses: CourseModel[] = [];
-  public filteredCourses: CourseModel[] = [];
   breadCrumbs: BreadCrumb[] = [{
     link: null,
     title: 'Courses'
   }];
   @ViewChild('modalView') private modalView;
+  public onSearchChanged: () => void;
+
+  public search = Observable.create((observer) => {
+    this.onSearchChanged = _.debounce(() => {
```

Comment:
I think, it would be better to use `Subject` (you may use `.next()` and `.subscribe()` methods) to throw new keys. Also you may process it with help of RxJS operators.

For example:
```
sbj = new Subject();

(keyup)=sbj.next($event);

sbj.pipe(
  debounceTime(1000),
  filter(...)
).subscribe(...)
```

More operators you may find there: https://rxjs-dev.firebaseapp.com/api